app/command.go

```diff
@@ -77,32 +77,33 @@ func (a *App) CreateCommandPost(post *model.Post, teamId string, response *model
 func (a *App) ListAutocompleteCommands(teamId string, T goi18n.TranslateFunc) ([]*model.Command, *model.AppError) {
 	commands := make([]*model.Command, 0, 32)
 	seen := make(map[string]bool)
+
+	for _, cmd := range a.PluginCommandsForTeam(teamId) {
+		if cmd.AutoComplete && !seen[cmd.Trigger] {
+			seen[cmd.Trigger] = true
+			commands = append(commands, cmd)
+		}
+	}
+
 	for _, value := range commandProviders {
 		if cmd := value.GetCommand(a, T); cmd != nil {
 			cpy := *cmd
-			if cpy.AutoComplete && !seen[cpy.Id] {
+			if cpy.AutoComplete && !seen[cpy.Trigger] {
 				cpy.Sanitize()
 				seen[cpy.Trigger] = true
 				commands = append(commands, &cpy)
 			}
 		}
 	}
 
-	for _, cmd := range a.PluginCommandsForTeam(teamId) {
-		if cmd.AutoComplete && !seen[cmd.Trigger] {
-			seen[cmd.Trigger] = true
-			commands = append(commands, cmd)
-		}
-	}
-
 	if *a.Config().ServiceSettings.EnableCommands {
 		teamCmds, err := a.Srv().Store.Command().GetByTeam(teamId)
 		if err != nil {
 			return nil, err
 		}
 
 		for _, cmd := range teamCmds {
-			if cmd.AutoComplete && !seen[cmd.Id] {
+			if cmd.AutoComplete && !seen[cmd.Trigger] {
```

Comment:
From your PR description, I understood that you also want to allow custom commands to be able to override as well. 
Should we iterate on these before or remove these conditions then ?

Reply:
Yes. Fixed.